internal/service/common/device.go

```diff
@@ -51,8 +51,9 @@ func ReplaceDeviceStatus(ctx context.Context, st store.Store, log logrus.FieldLo
 			return nil, err
 		}
 	}
-	if oldDevice.Status.Lifecycle.Status == api.DeviceLifecycleStatusEnrolled && device.Status.Lifecycle.Status == api.DeviceLifecycleStatusUnknown {
-		device.Status.Lifecycle.Status = api.DeviceLifecycleStatusEnrolled
+	// do not overwrite valid service-side lifecycle status with placeholder device-side status
+	if device.Status.Lifecycle.Status == api.DeviceLifecycleStatusUnknown {
+		device.Status.Lifecycle.Status = oldDevice.Status.Lifecycle.Status
```

Comment:
It seems to me that both the old and new code here is not necessary.
A couple lines down we call UpdateServiceSideStatus(), which calls updateServerSideLifecycleStatus(), which should set device.Status.Lifecycle the status to decommissioning if necessary.

Reply:
This logic is needed to update the data passed into the `updateServerSideLifecycleStatus()` function, otherwise there is a bootstrapping problem where the first "blank" `lifecycle status` created device-side overwrites the valid `enrolled` lifecycle status tracked service-side. The service-side owns the lifecycle status so it should be updated this way AFAIU.